scripts/pre_commit_linter.py

```diff
@@ -1197,6 +1197,53 @@ def _check_directive_scope(self):
 
         return summary_messages
 
+    def _check_js_file_overview(self):
+        """This function checks the JS files for the presence of their
+        respective file overviews.
+        """
+        if self.verbose_mode_enabled:
+            print 'Starting JS file overview check'
+            print '----------------------------------------'
+
+        summary_messages = []
+        failed = False
+        files_to_check = [
+            filepath for filepath in self.all_filepaths if (
+                filepath.endswith('.js'))
+            and not any(fnmatch.fnmatch(filepath, pattern) for pattern in
+                        EXCLUDED_PATHS)]
+
+        with _redirect_stdout(_TARGET_STDOUT):
+            for filepath in files_to_check:
+                parsed_script = self.parsed_js_files[filepath]
+                file_overview_found = False
+                for comment in parsed_script.comments:
+                    if '@fileoverview' in comment.value:
+                        file_overview_found = True
+                        if comment.type != 'Block':
+                            failed = True
+                            print ('%s --> File overview must be in block '
+                                'comments.' % filepath)
+                if not file_overview_found:
```

Comment:
I think this can be added as ```else``` or  ```elif``` statement after ```if '@fileoverview' in comment.value:```

Reply:
Actually, the `parsed_script.comments` is a list of all the comments it has extracted line by line in a single file. So since every file contains the copyright notice in the form of multiple single line comments, the list contains them too. And since these comments don't have the file overview, the code will end up in the else block, even without checking for the comment that contains the file overview.

Reply:
Okay thanks @YashJipkate !